tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,484 @@
+                if datetimeformattz not in ['YYYY-MM-DD HH:MI:SSOF',
+                                            'YYYY-MM-DD HH24:MI:SSOF']:
+                    # We will be using pandas
+                    addl_hints.update(pandas_compatible_addl_hints)
+                    if 'AM' in datetimeformattz:
+                        # TODO: Add a GitHub issue for this
+                        logger.warning('Cannot export this dateformat using Pandas or Vertica--'
+                                       'skipping test')
+                        continue
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'datetimeformattz': datetimeformattz,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            csv_text = self.unload(column_name='timestamptz',
+                                   records_format=records_format)
+            self.assertIn(csv_text, [create_sample(datetimeformattz) + "\n",
+                                     create_sample(datetimeformattz).replace('-00', '+00') + "\n",
+                                     create_sample(datetimeformattz).replace('-00', '') +
+                                     ".000000\n",
+                                     # TODO: Should this be necessary?
+                                     create_sample(datetimeformattz) + ".000000\n",
+                                     # TODO: Should this be necessary?
+                                     create_sample(datetimeformattz).replace('-00', '') +
+                                     '.000000+0000\n',
+                                     # TODO: Should this be necessary?
+                                     create_sample(datetimeformattz) +
+                                     f":{SAMPLE_SECOND:02d}.000000\n"
+                                     ],
+                          f"from datetimeformattz {datetimeformattz} and addl_hints {addl_hints}")
+
+    def test_unload_timeonly(self) -> None:
+        self.createTimeTable()
+        for timeonlyformat in TIMEONLY_CASES:
+            pandas_compatible_addl_hints: PartialRecordsHints = {
+                'dateformat': 'YYYY-MM-DD',
+                'datetimeformat': f'YYYY-MM-DD {timeonlyformat}',
+                'datetimeformattz': f'YYYY-MM-DD {timeonlyformat}',
+            }
+            addl_hints: PartialRecordsHints = {}
+            if self.engine.name == 'redshift' and not self.has_scratch_s3_bucket():
+                # Will be using Pandas
+                addl_hints = pandas_compatible_addl_hints
+                if 'AM' in timeonlyformat:
+                    # TODO: Add a GitHub issue for this
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker